src/rt/sched/behaviourcore.h

```diff
@@ -956,26 +995,29 @@ namespace verona::rt
         acquire_with_transfer(cown, transfer_count, ref_count);
 
         // Process writes without predecessor
-        if ((chain_had_no_predecessor) && (!curr_slot->is_read_only()))
+        if ((chain_had_no_predecessor) && (!chain_first_slot->is_read_only()))
         {
           if (cown->read_ref_count.try_write())
           {
             Logging::cout() << " Writer at head of queue and got the cown "
-                            << *curr_slot << Logging::endl;
+                            << *chain_first_slot << Logging::endl;
             ex_count++;
             yield();
           }
           else
           {
             Logging::cout() << " Writer waiting for previous readers cown "
-                            << *curr_slot << Logging::endl;
+                            << *chain_first_slot << Logging::endl;
             yield();
             cown->next_writer = first_body;
           }
         }
 
         // Process execution count
         ec[first_body_index] += ex_count;
+
+        for (int i = 1; i < first_consecutive_readers_count; i++)
+          ec[first_body_index + i] += 1;
```

Comment:
Why is `ex_count` not used here?  It feels like the whole chain should get to go if the read available was seen during scheduling?